requirements/constraints.txt

```diff
@@ -0,0 +1,17 @@
+# Version constraints for pip-installation.
+#
+# This file doesn't install any packages. It specifies version constraints
+# that will be applied if a package is needed.
+#
+# When pinning something here, please provide an explanation of why. Ideally,
+# link to other information that will help people in the future to remove the
+# pin when possible.  Writing an issue against the offending project and
+# linking to it here is good.
+
+Flask-Dance[sqla]>=0.9,<0.10
+celery>=3,<4
+github3.py>=0.9,<1.0
+redis<3.0       # 3.0 broke something: https://github.com/andymccurdy/redis-py/issues/1068
```

Comment:
there is a doc to specify exact version rather than the minimum number, please follow that.